mass-update-cache.py

```diff
@@ -0,0 +1,106 @@
+from process_pr import (
+    ISSUE_SEEN_MSG,
+    CMSBOT_TECHNICAL_MSG,
+    REGEX_COMMITS_CACHE,
+    loads_maybe_decompress,
+)
+
+setdefaulttimeout(120)
+
+SCRIPT_DIR = dirname(abspath(sys.argv[0]))
+
+
+# Notice: can't use `process_pr.read_bot_cache` since it modifies bot cache before returning.
+# Another solution is to add a flag to `read_bot_cache` to load cache without adding missing keys
+def load_bot_cache_local(comment_msg):
+    seen_commits_match = REGEX_COMMITS_CACHE.search(comment_msg)
+    if seen_commits_match:
+        print("Loading bot cache")
+        res = loads_maybe_decompress(seen_commits_match[1])
+        return res
+    return {}
+
+
+def main():
+    parser = ArgumentParser(usage="%prog <owner/repo>")
+    parser.add_argument("-o", "--organization", dest="org", default="cms-sw")
+    parser.add_argument("-n", "--dry-run", dest="dryrun", action="store_true")
+    args = parser.parse_args()
+
+    gh = github.Github(login_or_token=get_gh_token())
+    api_rate_limits(gh)
+
+    with open("runme.sh", "w") as f:
+        gh_org = gh.get_organization(args.org)
+        for repo in gh_org.get_repos():
+            print("Using repository", repo.full_name)
+            repo_name = repo.full_name
+            repo_dir = join(SCRIPT_DIR, "repos", repo_name.replace("-", "_"))
+            if exists(repo_dir):
+                sys.path.insert(0, repo_dir)
+            import repo_config
+
+            cmsbuild_user = repo_config.CMSBUILD_USER
+            set_gh_user(cmsbuild_user)
+
+            if not getattr(repo_config, "RUN_DEFAULT_CMS_BOT", True):
+                sys.exit(0)
+            repo = gh.get_repo(repo_name)
+
+            cnt = 0
+
+            for issue in repo.get_issues():
+                print("  Processing Issue#{0}: {1}".format(issue.number, issue.title))
+                bot_cache = None
+                for comment in issue.get_comments():
+                    if comment.user.login.encode("ascii", "ignore").decode() != cmsbuild_user:
+                        continue
+                    comment_msg = (
+                        comment.body.encode("ascii", "ignore").decode() if comment.body else ""
+                    )
+                    first_line = "".join(
+                        [line.strip() for line in comment_msg.split("\n") if line.strip()][0:1]
+                    )
+                    if re.match(ISSUE_SEEN_MSG, first_line):
+                        bot_cache = load_bot_cache_local(comment_msg)
+                        print("    Read bot cache from already seen comment:", comment)
+                        break
+                    elif re.match(CMSBOT_TECHNICAL_MSG, first_line):
+                        bot_cache = load_bot_cache_local(comment_msg)
+                        print("    Read bot cache from technical comment:", comment)
+                        break
+                if bot_cache and ("commits" not in bot_cache):
```

Comment:
I guess we would like to stop if there is no bot cache ... right? We do not want to process over 40K PRs , we only want to process those PRs where we have old bot cache